engine/access/rpc/backend/backend_scripts.go

```diff
@@ -4,381 +4,379 @@ import (
-			"script execution results on Archive node and EN are not equal")
-	}
-}
+		b.compareScriptExecutionResults(execResult, execErr, localResult, localErr, blockID, script, insecureScriptHash)
 
-func (b *backendScripts) logScriptExecutionComparison(
-	blockID flow.Identifier,
-	script []byte,
-	execNodeResult []byte,
-	archiveResult []byte,
-	executionError error,
-	archiveError error,
-	msg string,
-) {
-	// over-log for ease of debug
-	if executionError != nil || archiveError != nil {
-		b.log.Debug().Hex("block_id", blockID[:]).
-			Str("script", string(script)).
-			AnErr("execution_node_error", executionError).
-			AnErr("archive_node_error", archiveError).
-			Msg(msg)
-	} else {
-		b.log.Debug().Hex("block_id", blockID[:]).
-			Str("script", string(script)).
-			Hex("execution_node_result", execNodeResult).
-			Hex("archive_node_result", archiveResult).
-			Msg(msg)
+		// always return EN results
+		return execResult, execErr
+
+	default:
+		return localResult, localErr
 	}
 }
 
-// executeScriptOnAvailableArchiveNodes executes the given script for a blockID on all archive nodes available
-func (b *backendScripts) executeScriptOnAvailableArchiveNodes(
+// executeScriptLocally executes the provided script using the local execution state.
+func (b *backendScripts) executeScriptLocally(
 	ctx context.Context,
 	blockID flow.Identifier,
+	height uint64,
 	script []byte,
 	arguments [][]byte,
-	insecureScriptHash [16]byte,
+	insecureScriptHash [md5.Size]byte,
 ) ([]byte, error) {
-	var errors *multierror.Error
-	if len(b.archiveAddressList) > 0 {
-		startTime := time.Now()
-		for idx, rnAddr := range b.archiveAddressList {
-			rnPort := b.archivePorts[idx]
-			result, err := b.tryExecuteScriptOnArchiveNode(ctx, rnAddr, rnPort, blockID, script, arguments)
-			if err == nil {
-				// log execution time
-				b.metrics.ScriptExecuted(
-					time.Since(startTime),
-					len(script),
-				)
-				return result, nil
-			} else {
-				errCode := status.Code(err)
-				switch errCode {
-				case codes.InvalidArgument:
-					// failure due to cadence script, no need to query further
-					b.log.Debug().Err(err).
-						Str("script_executor_addr", rnAddr).
-						Hex("block_id", blockID[:]).
-						Hex("script_hash", insecureScriptHash[:]).
-						Str("script", string(script)).
-						Msg("script failed to execute on the execution node")
-					return nil, err
-				case codes.NotFound:
-					// failures due to unavailable blocks  are explicitly marked Not found
-					b.metrics.ScriptExecutionErrorOnArchiveNode()
-					b.log.Error().Err(err).Msg("script execution failed for archive node")
-					return nil, err
-				default:
-					errors = multierror.Append(errors, err)
-					continue
-				}
-			}
+	execStartTime := time.Now()
+
+	result, err := b.scriptExecutor.ExecuteAtBlockHeight(ctx, script, arguments, height)
+
+	execEndTime := time.Now()
+	execDuration := execEndTime.Sub(execStartTime)
```

Comment:
Could we submit this as a metric, it would be really useful to detect any issues.